### PR TITLE
add new task for generating static b by linking

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

Adds a new task for generating the static B matrix by linking. This will avoid the need to run the full task when a larger number of  nodes are needed and there is no dependency on the cycle.

### CODE REVIEW

File: src/swell/utilities/file_system_operations.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    # Check that at least one path was found
+    if not any(found_paths):
+        logger.abort(f'In link_all_files_from_first_in_hierarchy_of_sources none of the ' +
+                     f'directories being searched were found to exist. Directories: {source_paths}')
+
+    # Second sweep to establish if directories contain files
+    print('found_paths', found_paths)
+    found_files = False
+    for ind, source_path in enumerate(source_paths):
+        print("AAA", ind, source_path, found_paths[ind])
```
Comment (Dooruk) [inline src/swell/utilities/file_system_operations.py]: I'm guessing these print statements were for testing?

File: src/swell/tasks/generate_b_climatology_by_linking.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            All inputs are extracted from the JEDI experiment file configuration.
+            See the taskBase constructor for more information.
+        """
+
+        # Get the flavor of static background error model
+        static_background_error_model = self.config_get('static_background_error_model')
+
+        # Extract general parts of the config
+        swell_static_files_main = self.config_get('swell_static_files')
+        swell_static_files_user = self.config_get('swell_static_files_user', None)
```
Comment (Dooruk) [inline src/swell/tasks/generate_b_climatology_by_linking.py:41]: Is this defined in any of the suites right now?

**Review by Dooruk — CHANGES_REQUESTED** [review]
It works on my end, just a few print statements left.

Also `swell_static_files_user` is not defined in any of the suites.

**Comment by danholdaway** [comment]
I'm thinking that `swell_static_files_user` will be a highly customized thing that folks can manually add to their YAML files.

File: src/swell/utilities/file_system_operations.py (hunk 1 above)
Comment (danholdaway) [reply src/swell/utilities/file_system_operations.py]: Oops! Thanks.

File: src/swell/tasks/generate_b_climatology_by_linking.py (hunk 2 above)
Comment (danholdaway) [reply src/swell/tasks/generate_b_climatology_by_linking.py:41]: Note that it defaults to `None` if the user does not provide this in their YAML. Generally I don't expect people to be using this much.

Comment (Dooruk) [reply src/swell/tasks/generate_b_climatology_by_linking.py:41]: I see. I guess my point is that people won't see it if it's not shown in or documentation.

Comment (danholdaway) [reply src/swell/tasks/generate_b_climatology_by_linking.py:41]: It's a good point. I guess the best thing would be to use R2D2 for these kinds of things.

**Pull request merged**